tests/buildpomtest.py

```diff
@@ -32,11 +32,16 @@ def test_parse_BUILD_pom(self):
         self.assertIs(pomgenmode.DYNAMIC, art_def.pom_generation_mode)
         self.assertEqual(None, art_def.pom_template_file)
         self.assertTrue(art_def.include_deps)
+        self.assertTrue(art_def.change_detection)
         self.assertEqual(package_rel_path, art_def.bazel_package)
         self.assertEqual(None, art_def.released_version)
         self.assertEqual(None, art_def.released_artifact_hash)
         self.assertIsNotNone(art_def.version_increment_strategy)
 
+        self._write_build_pom_set_change_detection(repo_package, artifact_id, group_id, version, "dynamic", False)
```

Comment:
@dylanShark just make this a different test please (new test method).  Since we can, lets add 2 new test methods.  One that sets change_detection explicitly to True and one that sets change_detection explicitly to False.

Reply:
Will do!